container.go

```diff
@@ -750,3 +774,45 @@ func (c *Container) detachDevices() error {
 
 	return nil
 }
+
+func (c *Container) addResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot add vCPUs to sandbox containers")
+			return nil
+		}
```

Comment:
I think this check must not be done. The sandbox type will never trigger a call to `createContainer()` unless there is a real container to create, and in that case, we want to hotplug the right amount of resources. By doing this, you are preventing the Docker case from working properly, meaning it will run only on default number of VCPUs.

Reply:
currently in docker the vCPUs are cold plugged, this code is to prevent adding double resources

Reply:
Well this should not be an issue. The docker case will start the VM with the default CPUs, and when the container (right after) will be created/added, the code should evaluate if it needs more/less CPUs. Why do we need to make a particular case here ?